app/src/main/java/com/spascoding/taskycourse/feature_register_screen/domain/repository/AuthRepository.kt

```diff
@@ -7,5 +7,5 @@ interface AuthRepository {
     suspend fun register(name: String, email: String, password: String): Response<Void>
     suspend fun login(email: String, password: String): Response<LoginResponse>
     suspend fun authenticate(token: String): Boolean
-    fun logout()
+    suspend fun logout(token: String)
```

Comment:
I think I mentioned this before, but why would the logout function need to know the token here? It can simply read that from preferences before making the call.